pkg/chunk/grpc/grpc_client.go

```diff
@@ -0,0 +1,25 @@
+package grpc
+
+import (
+	"flag"
+
+	"google.golang.org/grpc"
+)
+
+// Config for a StorageClient
+type Config struct {
+	Address string `yaml:"address,omitempty"`
+}
+
+// RegisterFlags adds the flags required to config this to the given FlagSet
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&cfg.Address, "grpc-store.server-address", "", "Hostname or IP of gRPC instance.")
+}
+
+func connectToGrpcServer(serverAddress string) (GrpcStoreClient, *grpc.ClientConn, error) {
```

Comment:
Note: I think using the gRPC client pool with the Thanos DNS service discovery (like we did [here](https://github.com/cortexproject/cortex/blob/master/pkg/querier/blocks_store_balanced_set.go)) would have been preferrable here, but **this can be left for a future iteration**, so no need to reiterate on it in this PR.